src/yang/ietf-babel.yang

```diff
@@ -954,13 +985,16 @@ module ietf-babel {
 
           action test {
             description
-              "An operation that allows the MAC key and hash
+              "An operation that allows the MAC key and MAC
                algorithm to be tested to see if they produce an
-               expected outcome. Input to this operation is a
-               binary string. The implementation is expected to
-               create a hash of this string using the value and
-               the algorithm. The output of this operation is
-               the resulting hash, as a binary string.";
+               expected outcome. Input to this operation are a
```

Comment:
Not sure if you mean that both the parameters and the descriptions need to be updated. Anyway, take a look at the updated model to let me know if it captures your sentiments.

Reply:
I think the input mac description is ambiguous wrt "calculated". How about: 
                   Input to this operation includes a MAC.
                   The implementation is expected to calculate
                   a MAC over the test-string using the value and
                   algorithm of this key object and
                   compare its calculated MAC to this input MAC.

Binary can be any length of 0's and 1's (and length must be specified). Boolean is effectively a binary of length 1 (and is usually stored that way in a datastore), but it's defined specially (most everywhere) because of its, well, binary specialness. Everywhere I'm familiar, Boolean is the preferred datatype for this sort of thing. BTW, I'll be naming the output parameter "Success" in TR-181.

Reply:
Updated per the comments.